### ISSUE TITLE
Mismatch results when using "OR" operator in sum_over_time function

### ISSUE THREAD

**Issue by groobyming** [VictoriaMetrics/VictoriaMetrics#3996]
### Describe the bug

When I use "OR" operator in sum_over_time,  got a mismatching result for related metrics
eg:
MetricsQL：
sum(sum_over_time(ingressctrl_api_request_total_value{cluster_id='K8S-41001'}[1m])) by (cluster_id, handler)

Result：
<img width="1700" alt="image" src="https://user-images.githubusercontent.com/34335395/227140179-370c2949-eada-4b1a-933b-10f1898fa229.png">

But when I use "OR" operator in sum_over_time, the result is as below, not expected
eg:
MetricsQL：
sum( sum_over_time(ingressctrl_api_request_total_value{cluster_id='K8S-41001'}[1m] or ingressctrl_api_request_total_value{cluster_id='K8S-40999'}[1m])) by (cluster_id, handler)

Result:
<img width="1699" alt="image" src="https://user-images.githubusercontent.com/34335395/227140785-ac8b93f2-624f-480c-9aba-a2ed56d28ae0.png">

### To Reproduce

Add "OR" operator in sum_over_time
eg:
sum( sum_over_time(ingressctrl_api_request_total_value{cluster_id='K8S-41001'}[1m] or ingressctrl_api_request_total_value{cluster_id='K8S-40999'}[1m])) by (cluster_id, handler)

### Version

v1.82.1-cluster

**Comment by hagen1778** [comment]
Hello!

When you use `or` or anything else except [series selector](https://docs.victoriametrics.com/keyConcepts.html#filtering) inside function - it gets [automatically transform](https://docs.victoriametrics.com/MetricsQL.html#implicit-query-conversions) into a subquery. Please read more about what is [a subquery](https://docs.victoriametrics.com/MetricsQL.html#subqueries) and how they are calculated. 

tl;dr; it is likely doing not what you expect. You probably need to run the following query:
```
sum(
sum_over_time(ingressctrl_api_request_total_value{cluster_id='K8S-41001'}[1m])
or 
sum_over_time(ingressctrl_api_request_total_value{cluster_id='K8S-40999'}[1m])
) by (cluster_id, handler)
```

**Comment by groobyming** [comment]
> Hello!
> 
> When you use `or` or anything else except [series selector](https://docs.victoriametrics.com/keyConcepts.html#filtering) inside function - it gets [automatically transform](https://docs.victoriametrics.com/MetricsQL.html#implicit-query-conversions) into a subquery. Please read more about what is [a subquery](https://docs.victoriametrics.com/MetricsQL.html#subqueries) and how they are calculated.
> 
> tl;dr; it is likely doing not what you expect. You probably need to run the following query:
> 
> ```
> sum(
>   sum_over_time(ingressctrl_api_request_total_value{cluster_id='K8S-41001'}[1m])
>   or 
>   sum_over_time(ingressctrl_api_request_total_value{cluster_id='K8S-40999'}[1m])
> ) by (cluster_id, handler)
> ```

@hagen1778 Thanks, it worked for me
